tests.c

```diff
@@ -53,6 +62,277 @@ test_opcode_0x13(void) // NOLINT
+
+  /* carry test */
+  // setup
+  cpu.h = 0x00; // NOLINT
+  cpu.l = 0x80; // NOLINT
+
+  // execute
+  code_found = execute_instruction(&cpu, 0x29); // NOLINT
+
+  // verify
+  CU_ASSERT(0 == code_found);
+  CU_ASSERT(2 == cpu.pc);
+  CU_ASSERT(0x00 == cpu.h); // NOLINT
+  CU_ASSERT(0x00 == cpu.l); // NOLINT
+  CU_ASSERT(FLAG_CY == (cpu.flags & FLAG_CY));
+}
+
+void
+test_opcode_0x3a(void) // NOLINT
+{                      // LDA adr
+
+  // setup
+  i8080 cpu;
+  cpu_init(&cpu);
+  cpu_write_mem(&cpu, 0x0001, 0xAB); // NOLINT
+  cpu_write_mem(&cpu, 0x0002, 0xCD); // NOLINT
+  cpu_write_mem(&cpu, 0xcdab, 0x42); // NOLINT
+
+  // execute
+  int code_found = execute_instruction(&cpu, 0x3a); // NOLINT
+
+  // verify
+  CU_ASSERT(0 == code_found);
+  CU_ASSERT(3 == cpu.pc);
+  CU_ASSERT(0x42 == cpu.a); // NOLINT
+
+  // cleanup
+  cpu_write_mem(&cpu, 0x0001, 0x00); // NOLINT
+  cpu_write_mem(&cpu, 0x0002, 0x00); // NOLINT
+  cpu_write_mem(&cpu, 0xcdab, 0x00); // NOLINT
+}
+
+void
+test_opcode_0x66(void) // NOLINT
+{                      // MOV H,M
+
+  // setup
+  i8080 cpu;
+  cpu_init(&cpu);
+  cpu_write_mem(&cpu, 0x1234, 0x31); // NOLINT
+  cpu.h = 0x34;                      // NOLINT
+  cpu.l = 0x12;                      // NOLINT
+
+  // execute
+  int code_found = execute_instruction(&cpu, 0x66); // NOLINT
+
+  // verify
+  CU_ASSERT(0 == code_found);
+  CU_ASSERT(1 == cpu.pc);
+  CU_ASSERT(0x31 == cpu.h); // NOLINT
+
+  // cleanup
+  cpu_write_mem(&cpu, 0x1234, 0x00); // NOLINT
+}
+
+void
+test_opcode_0x7b(void) // NOLINT
+{                      // MOV A,E
+
+  // setup
+  i8080 cpu;
+  cpu_init(&cpu);
+  cpu.e = 0x29; // NOLINT
+
+  // execute
+  int code_found = execute_instruction(&cpu, 0x7b); // NOLINT
+
+  // verify
+  CU_ASSERT(0 == code_found);
+  CU_ASSERT(1 == cpu.pc);
+  CU_ASSERT(0x29 == cpu.a); // NOLINT
+  CU_ASSERT(0x29 == cpu.e); // NOLINT
+}
+
+void
+test_opcode_0xaf(void) // NOLINT
+{                      // XRA A
+
+  // setup
+  i8080 cpu;
+  cpu_init(&cpu);
+  cpu.a = 0xcc; // NOLINT
+
+  // execute
+  int code_found = execute_instruction(&cpu, 0xaf); // NOLINT
+
+  // verify
+  CU_ASSERT(0 == code_found);
+  CU_ASSERT(1 == cpu.pc);
+  CU_ASSERT(0x00 == cpu.a); // NOLINT
+  CU_ASSERT(FLAG_S != (cpu.flags & FLAG_S));
+  CU_ASSERT(FLAG_Z == (cpu.flags & FLAG_Z));
+  CU_ASSERT(FLAG_P == (cpu.flags & FLAG_P));
+  CU_ASSERT(FLAG_CY != (cpu.flags & FLAG_CY));
+  CU_ASSERT(FLAG_AC != (cpu.flags & FLAG_AC));
+}
+
+void
+test_opcode_0xc5(void) // NOLINT
+{                      // PUSH B
+
+  // setup
+  i8080 cpu;
+  cpu_init(&cpu);
+  cpu.b = 0xcd;    // NOLINT
+  cpu.c = 0xab;    // NOLINT
+  cpu.sp = 0xcccc; // NOLINT
+
+  // execute
+  int code_found = execute_instruction(&cpu, 0xc5); // NOLINT
+
+  // verify
+  CU_ASSERT(0 == code_found);
+  CU_ASSERT(1 == cpu.pc);
+  CU_ASSERT(0xccca == cpu.sp); // NOLINT
+  CU_ASSERT(cpu.b == cpu_read_mem(&cpu, cpu.sp + 1));
+  CU_ASSERT(cpu.c == cpu_read_mem(&cpu, cpu.sp));
+}
```

Comment:
Missing clean up at the end of the test. the op code writes to cpu memory so that will need to be zeroed out.